touch-core-internal/Services/EmployeeRepository.cs

```diff
@@ -21,17 +21,22 @@ public EmployeeRepository(IMapper mapper, DataContext dataContext)
 
         public IMapper Mapper { get; private set; }
 
-        public async Task<ServiceResponse<List<GetEmployeeDTO>>> AddNewEmployeeAsync(AddEmployeeDTO newEmployee)
+        public async Task<ServiceResponse<GetEmployeeDTO>> AddNewEmployeeAsync(AddEmployeeDTO newEmployee)
         {
-            var serviceResponse = new ServiceResponse<List<GetEmployeeDTO>>();
+            var serviceResponse = new ServiceResponse<GetEmployeeDTO>();
+
+            if (await this.GetEmployeeByEmailAsync(newEmployee.Email) != null)
```

Comment:
Have you tested this scenario? `await this.GetEmployeeByEmailAsync(newEmployee.Email)` returns `ServiceResponse<GetEmployDTO>` already you are creating it's instance in this function. I guess you need to check `.Data` instead of `ServiceResponse<GetEmployeeDTO>`. Correct me if I am wrong here

Reply:
Good catch. Update with proper check